### PR TITLE
UCS: Config parser enhancements

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/ucs/config/parser.h [hunk 1]
```diff
@@ -75,36 +75,38 @@ typedef struct ucs_range_spec {
...
 
 typedef struct ucs_config_global_list_entry {
-    ucs_list_link_t          list;
-    const char               *name;
-    const char               *prefix;
-    ucs_config_field_t       *fields;
-    size_t                   size;
+    const char               *name;    /* configuration table name */
+    const char               *prefix;  /* configuration prefix */
+    ucs_config_field_t       *table;  /* array of configuration fields */
```
Comment (hoopoepg) [inline src/ucs/config/parser.h]: minor: unaligned comment

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11414/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 33 workers (click for details)</summary>

**Note:** the logs will be deleted after 28-Jul-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/68561a6c26ce188697c77b0a314d2465458747ae/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/754d7a6c26a3d7a33abd368f7e59dc0bd2160e3d/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/f470638ff069904910bd6ddc7c92a0c7f2ffd9bd/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/68e03aa7e795685f73db9f3b89391c8e6fd23d8a/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/6244a26a083149769dacd6b30803cc5e690aa378/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/deb80bccee9c6eae12b439ee77f36e2858029fb1/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/efc3cf8c94c6ae0d5d07e4146f438835ceb8ebae/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/9db944d69c69808744674aa438366b04479d23ca/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/17c2fc102de0bf6dc3e718147b0cc1ab277e0459/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/b6f82482f6c937117dd884ebae831d1305e10c40/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/525f9add1613bc30fca7e318669e6724dc01cb89/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/363965d032f70b15ae85b5a94a811e12d26c8cf4/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/49a6239a06b8be0aef5e50160a46f01c4c7fe577/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/d7a5de8a42aeb968b52a350b2abcf487bfaa1c62/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/d50a73882c158813d2570cbc20db9c77a8c0847d/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/f43d92054ee336e3b1e3c2947cfcafdfccb2756c/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/984951296a6f4adc1a7992555ab60a5aae2bcc36/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/f4866e5a50c7972d51fdec6ca677bbcd7666b3f9/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/dd89fa5a5cc751bf01e49dbc17e382a138070674/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/1b9e3d279cb28c70a91d8f421d6b575569f3d5b6/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/b45c848f558a77f4681032d0563a25ed6d04a7e7/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/50694cfad61c52977a04597bcdfe143e7993ab11/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/7680cf0f7e429eb9929b8bf07917d7fefc364043/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/11257f20984311ebef82d72b6345c9a706f7a806/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/2a3b90969200467b5a0ffebe0f944450d1df9c38/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W0](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/b79f4d8b1906993b31521e77f1cbd848b95c4179/hpc-test-node-upstream_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W1](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/7fa73566ed17389f6fcddf0ad3dfdf3df3057ad3/hpc-test-node-upstream_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W2](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/2d2fd027dc864581d415aa7010dca914874b26e0/hpc-test-node-upstream_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W3](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/da0466580ecb88d79396ba562b6e8acd4a10b78d/hpc-test-node-upstream_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/99cfcc30c4726e101c8e52476c9eb6d11261bea4/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/f7c593190c526a5573c82266ab8c10691d1aec20/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/a2e668d91c0fab6af79d830d83f2a0aa6944f76e/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/65740aa9c1a4f75a456eed527295c974/raw/b312c7c3d70353d87fd20a79d51a7a66127e8f96/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11467/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 33 workers (click for details)</summary>

**Note:** the logs will be deleted after 30-Jul-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/e0413b1696347740a99b1bfd3a33a5b87b47da95/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/d01a8008a58a550e3d8048d5a5894b33fb1a49c4/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/96deb1cd5f789d4eb28de18d02b5d8c7e8f2abd0/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/9e3ccce4a9ca92f1288b64ca87523834955faf42/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/57235cd79a9c28ba1db8d941cd0d683a5848bd23/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/c8af9bfa99129b302e5bb9daa76522576714813e/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/bf84536dbc0a003bd8e232a09d08d522c6e69bdf/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/18f590c55dccde5b6d03548ff7884e32ef6f8ec7/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/cb561b69f6978e2d042d8eeeac723ea21dc985cc/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/7bcc48b6d30d70972c5bd12b5d4b58044dda7a10/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/0af23548ce9d8cc8bf645e846a2378eae23db5dd/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/eb93118165762c540482097c27ca2d1d17ed1acf/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/368efed933d3d732833660de7e1ef06c87a8d1b8/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/190e8c4c35ade32f14d6c01c98f3eb0b2fde816c/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/322bff5cbc9f5454a8cd0487b8852acdfe469109/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/7d0a802dc5259ccccb508c2f54f40eeade88a9dd/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/cedc5daccc81800e13cefdd5e4351f82dd1ce0c5/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/f80938f68147eaa8fdac3640405c1a6f17d49507/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/510c45bb67cde491802acd28cc5c560d1a6f0a08/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/99bb75d8dc0cf45a47ae3c385c22b3cb49c785b5/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/a1e64fc910b7516db25f19c9c16187eefd3a071e/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/b6d787f0506648db95ac3f58e8bcd4e558fd05d7/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/d9cae543174873d5d6a3e1a77108494652f2b46f/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/693cb97b8c02abefac8a08607af4815e9e35e97a/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/c394f20d0f2eefc8a7498eef878da28bf84ef396/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W0](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/ac40a54942bf751ab2fb59eccd56afdcf935c96f/hpc-test-node-upstream_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W1](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/54142e40c6719cee4fa84d2078000d302e3a5a41/hpc-test-node-upstream_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W2](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/dc82281a80eb246328e51f3a986a9aead7b83013/hpc-test-node-upstream_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W3](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/2222d75fba2301f4394e1ef12a0e0ce01a42f396/hpc-test-node-upstream_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/a9148cc9e41bfdfae0c72dd77f4f2ca808b631da/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/47f8e205d0c6dfb85bfbd7b8e55c052b420ce9c6/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/90b0c13270e66117842c297bfd1188247b8b7107/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/fff194792ed90846f4ab8f89f43b0591/raw/247f8dd08b022ba3094a26be859ea21e23591269/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**